web/packages/shared/components/Validation/rules.ts

```diff
@@ -107,28 +99,88 @@ const requiredConfirmedPassword =
     };
   };
 
-const isIamRoleNameValid = roleName => {
+/**
+ * DNS1035_LABEL_REGEX uses the same regex matcher as documented here:
+ * https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#rfc-1035-label-names
+ *
+ * This matches how the backend validates integration names to satisfy a lowercase valid DNS subdomain.
+ */
+const DNS1035_LABEL_REGEX = /^[a-z]([-a-z0-9]*[a-z0-9])?$/;
+const DNS1035_LABEL_MAX_LENGTH = 63;
+const isIntegrationNameValid = (integrationName: string) => {
   return (
-    roleName && roleName.length <= 64 && roleName.match(IAM_ROLE_NAME_REGEX)
+    integrationName &&
+    integrationName.length <= DNS1035_LABEL_MAX_LENGTH &&
+    integrationName.match(DNS1035_LABEL_REGEX)
   );
 };
 
+/**
+ * IAM_ROLE_NAME_REGEX uses the same regex matcher used in the backend:
+ * https://github.com/gravitational/teleport/blob/master/api/utils/aws/identifiers.go#L178
+ *
+ * The regex checks for alphanumerics and select few characters.
+ */
+const IAM_ROLE_NAME_REGEX = /^[\w+=,.@-]+$/;
+const IAM_ROLE_NAME_MAX_LENGTH = 64;
+const isIamRoleNameValid = (roleName: string) => {
+  return (
+    roleName &&
+    roleName.length <= IAM_ROLE_NAME_MAX_LENGTH &&
+    roleName.match(IAM_ROLE_NAME_REGEX)
+  );
+};
+
+/**
+ * ROLES_ANYWHERE_NAME_REGEX uses the same regex matcher used in the backend:
+ * https://github.com/gravitational/teleport/blob/master/api/utils/aws/identifiers.go#L213
```

Comment:
nit: Consider linking to a commit sha (or omitting the line number) as this can change in master.